packages/palette/src/utils/useResizeObserver.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * Adapted from https://github.com/jaredLunde/react-hook/blob/3c813dab5b21e26f2c85e733314ca5c063c6bfce/packages/resize-observer/src/index.tsx
```

Comment:
Primarily because I like this approach of re-using a single `ResizeObserver`: https://groups.google.com/a/chromium.org/g/blink-dev/c/z6ienONUb5A/m/F5-VcUZtBAAJ